docs/DeveloperGuide.md

```diff
@@ -95,6 +95,21 @@ The sequence diagram below illustrates a more in-depth view of the interactions
 
 <puml src="diagrams/UiSequenceDiagram.puml" alt="Sequence Diagram of UI Component"/>
 
+#### **Considerations for UI**
+
+##### Dynamic UI Updates
+
+The UI is designed to update dynamically based on changes in the `Model`. We narrowed down to two design choices for updating the UI components. They are:
+
+1. **Update using listeners embeded into UI components** - This design choice would involve embedding listeners into the UI components that would listen for changes in the `Model` (e.g. adding a listener to filteredPersons in ExamListPanel). This would allow for a more loosely coupled system, but would involve more complex implementation which could get messy as the number of listeners increase.
+2. **Update using a centralized update method** - This design choice involves having a centralized `update` method in the `MainWindow` that would call an `update` method in all other UI components after every command. This would involve a more tightly coupled system and may involve unnecessary updates, but would be easier to implement and maintain.
+
+We chose the second design choice as having a centralized update method would allow for easier maintenance, as there is a clear indicator of how UI components are updated from `MainWindow`. Adding extensions would also be more straightforward as future developers would know where to look for the update logic.
```

Comment:
Great justification on why we chose the centralized method! Keeps our UI error free and foolproof.